packages/@o3r/schematics/src/rule-factories/ng-add/dependencies.ts

```diff
@@ -0,0 +1,177 @@
+import { chain, Rule, Schematic } from '@angular-devkit/schematics';
+import { NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import * as path from 'node:path';
+import type { PackageJson } from 'type-fest';
+import * as semver from 'semver';
+import { getPackageManager, getWorkspaceConfig, SupportedPackageManagers } from '../../utility';
+import { NodePackageInstallTask, RunSchematicTask } from '@angular-devkit/schematics/tasks';
+
+/**
+ * Options to be passed to the ng add task
+ */
+export interface NgAddSchematicOptions {
+  /** Name of the project */
+  projectName?: string | null;
+
+  /** Skip the run of the linter*/
+  skipLinter?: boolean;
+
+  /** Skip the install process */
+  skipInstall?: boolean;
+
+  [x: string]: any;
```

Comment:
should not be optional this one too ?

Reply:
It is thanks to `[x: string]` :).
The purpose is just to allow additional fields, they are not mandatory

Reply:
ok. my bad. I was thinking it forces you to have at least one field :)